low-level/actuators/impl/generic/realstor_encl.py

```diff
@@ -608,7 +641,95 @@ def _get_encl_response(self, uri, request_type):
+            action_type = ctrl_action
+        else:
+            ctrl_cmd = f"{ctrl_action} {ctrl_type} {ctrl_name}"
+            action_type = f"{ctrl_action} {ctrl_type}"
+
+        if (ctrl_action == "restart" and not resource.endswith(('sc', 'mc', "*"))) \
+            or (ctrl_action == "shutdown" and resource.endswith(('sc', 'mc'))) or \
+            not ctrl_cmd.endswith((' a', ' b', ' both')):
+            invalid_args = True
+
+        if invalid_args:
+            # Invalid resource 'shutdown abc' for an
+            # 'ENCL: enclosure:fru:controller:shutdown' actuator request
+            err_msg = "Invalid resource '{}' for an '{}' actuator request".format(
+                resource, enclosure_request)
+            logger.error(err_msg)
+            raise Exception(err_msg)
+
+        encl_response = self._get_encl_response(
+            self.rssencl.URI_CLIAPI_BASE + ctrl_cmd.replace(' ', '/'),
+            self.rssencl.ws.HTTP_GET
+        )
+        if encl_response == {}:
+            severity = "warning"
+            message = "request failed. Please try again...!"
+
+        # Some info will be deleted from the "response_str" in the "make_response"
+        # method which is not required further. We are keeping this to decide some
+        # fields in the "make_response" method.
+        response_str = {
+            'shutdown' : {
+                'message': 'Shutdown %s' % (message),
+                'description': 'Shuts down the Storage Controller in a controller '
+                    'module. This ensures that a proper failover sequence is used, '
+                    'which includes stopping all I/O operations and writing any '
+                    'data in write cache to disk.',
+                'command': ctrl_cmd,
+                'alert_type': 'control:shutdown',
+                'severity': severity,
+                'resource_type': 'enclosure:fru:controller'
+                },
+            'restart sc' : {
+                'message': 'Restart / Start Storage Controller %s' % (message),
+                'description' : 'Restarts the Storage Controller in a controller '
+                    'module. When you restart a Storage Controller, it attempts '
+                    'to shut down with a proper failover sequence, which includes '
+                    'stopping all I/O operations and flushing the write cache to '
+                    'disk, and then the Storage Controller restarts. Restarting a '
+                    'Storage Controller restarts the corresponding Management '
+                    'Controller.',
+                'command': ctrl_cmd,
+                'alert_type': 'control:restart',
+                'severity': severity,
+                'resource_type': 'enclosure:fru:controller:sc'
+                },
+            'restart mc' : {
+                'message' : 'Restart / Start Management Controller %s' % (message),
+                'description' : 'Restarts the Management Controller in a '

```

Comment:
Good you captured this description(impact) from api doc, serves good.

Reply:
Yes